Updating Daft links in Ray documentation

ccmao1130 opened the pull request:
<!-- Thank you for your contribution! Please review https://github.com/ray-project/ray/blob/master/CONTRIBUTING.rst before opening a pull request. -->

<!-- Please add a reviewer to the assignee section when you create a PR. If you don't have the access to it, we will shortly find a reviewer and assign them to your PR. -->

## Why are these changes needed?

Want to update Daft links, messaging, and logo across Ray documentation

## Related issue number

n/a

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a
           method in Tune, I've added it in `doc/source/tune/api/` under the
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

aslonnie committed:
[data] adapt dask on ray to the new dask task class  (#54108)

"Dask on Ray" (DoR) is broken in dask==2024.11.0 or later as reported in
https://github.com/ray-project/ray/issues/48689 because Dask removed a
private function in https://github.com/dask/dask/pull/11378 that DoR has
been relying on. Not only https://github.com/dask/dask/pull/11378, Dask
has migrated their task data structure to a new format (the high-level
motivation is described in https://github.com/dask/dask/issues/9969).
Since this migration spans across a series of changes between 2024.11.0 and
2025.1.0, it's not realistic to copy what's been removed and paste them
in Ray.

This change adapts Dask on Ray to the change to keep its functionality.
The change is compatible only with
`dask>=2024.11.0,<2025.1.0` because Dask made another major change in
2025.1.0, breaking the shuffle optimization introduced in
https://github.com/ray-project/ray/pull/13951

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Co-authored-by: Hiromu Hota <hiromu.hota@gmail.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

ccmao1130 committed:
update daft link & blurb

Signed-off-by: ccmao1130 <chanchanmao1130@gmail.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

ccmao1130 committed:
update logo

Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

edoakes committed:
[core] Use `SignalActor` in `test_hybrid_policy_threshold` (#54312)

The semaphore is clever, but using signal instead for consistency with
other tests.

---------

Signed-off-by: Edward Oakes <ed.nmi.oakes@gmail.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

dayshah committed:
[core] Skip generator reconstruction test (#54320)

https://github.com/ray-project/ray/pull/53999 resulted in this test
being flaky on mac. This test's purpose seems to be similar to
https://github.com/ray-project/ray/blob/986115ce566fda437c5e3fcca3705c225b06f3b8/python/ray/tests/test_streaming_generator_4.py#L73
and was kind of trying to test a feature that didn't exist. But since it
wasn't actually pausing the generator for backpressure, the generator
would usually finish before the node removal actually happens. Now
sometimes when the node removal happens before the generator finishes,
we'll lose objects and go through the new path. We could also go through
the new resubmission path multiple times for one node death because
multiple objects from the same generator may be marked lost. Therefore,
sometimes we run out of retries before getting to the third retry in the
test and it fails with
`ray.exceptions.RayTaskError(ObjectReconstructionFailedMaxAttemptsExceededError)`

The fix to make this not flaky would be to do the follow up listed in
the previous pr.
> Currently, if multiple objects from the same generator are queued up
to be recovered when the recovery periodical runner runs, we could
resubmit for the first object and then once again queue up a resubmit
for the second if argument resolution and sequence numbering lines up.
Since this doesn't actually affect correctness and requires a bit of
refactoring, it'll be in a follow-up PR.

---------

Signed-off-by: dayshah <dhyey2019@gmail.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

davidxia committed:
[doc][kuberay] state `rayStartParams` is optional starting with KubeRay 1.4.0 (#53943)

but that if autoscaling is used, the autoscaler image must have Ray
2.45.0 or later.

closes https://github.com/ray-project/kuberay/issues/3580

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git
commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for
https://docs.ray.io/en/master/.
- [ ] I've added any new APIs to the API Reference. For example, if I
added a
method in Tune, I've added it in `doc/source/tune/api/` under the
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a
few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

Signed-off-by: David Xia <david@davidxia.com>
Co-authored-by: angelinalg <122562471+angelinalg@users.noreply.github.com>
Co-authored-by: Dhyey Shah <dhyey2019@gmail.com>
Co-authored-by: Kai-Hsun Chen <kaihsun@anyscale.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

dayshah committed:
[core] Fix gcs register actor callback check (#53634)

## Problem

This check assumes that if the actor is not in `registered_actors_` it
must be in `actor_to_register_callbacks_`.
https://github.com/ray-project/ray/blob/c6d7d7eaa1e4dd0fd42ba45891d8501ab14ceb44/src/ray/gcs/gcs_server/gcs_actor_manager.cc#L871-L872

This isn't true because in `DestroyActor`, the actor is always removed
from `actor_to_register_callbacks_`,

https://github.com/ray-project/ray/blob/c6d7d7eaa1e4dd0fd42ba45891d8501ab14ceb44/src/ray/gcs/gcs_server/gcs_actor_manager.cc#L1073
but only removed from `registered_actors_` if the actor is restartable.

https://github.com/ray-project/ray/blob/c6d7d7eaa1e4dd0fd42ba45891d8501ab14ceb44/src/ray/gcs/gcs_server/gcs_actor_manager.cc#L1143-L1146

It's also erasing from `actor_to_register_callbacks_` without actually
calling the callbacks, so rpc's from the core worker side could be left
hanging forever.

## Fix
The fix is to never erase from `actor_to_register_callbacks_` in
`DestroyActor`, and to always respond to the all the queued callbacks
with the appropriate status in the Put callback. The logic for which
status to respond with is the same. If it's in `registered_actors_`
after the table put is done the rpc should be completed with the ok
status. If it's not there by the time the table put is done it, it
should respond to the rpc with SchedulingCancelled.

Additional changes
- removed the need to pass the actor ptr into the RegisterActor callback
because it was unused
- removed an accessor that was only used for tests, added it to the test
fixture and turned the test fixture into a friend
- add logic in the accessor to read both the gRPC status and the
GcsStatus see comment here
https://github.com/ray-project/ray/pull/53634#issuecomment-3029076719

### Follow ups
- Fix the mess noted here
https://github.com/ray-project/ray/pull/53634#issuecomment-3029076719
- There's almost surely more lurking issues here due to actor management
split brain + kv operation ordering assumptions. Needs to be
investigated.
- The actor state transition machine here needs to be clearer, e.g.
actors shouldn't be put into registered_actors_ if they're not
registered yet.

---------

Signed-off-by: dayshah <dhyey2019@gmail.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

ccmao1130 committed:
update daft url

Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

ccmao1130 committed:
update daft logo

Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

zcin committed:
Fix `test_runtime_env_container` (#54330)

Move `httpx` out of `test_utils` because for some reason it is not
available in the image used for `test_runtime_env_container.

Signed-off-by: Cindy Zhang <cindyzyx9@gmail.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

can-anyscale committed:
[core][task-manager/02] consolidate TaskManager interface (#54317)

This PR is smaller than it looks.

The `TaskManager` class currently exposes two interfaces: `TaskFinisher`
and `TaskResubmission`. While these interfaces are well-intentioned,
they are only implemented by `TaskManager` itself, and the methods they
define are not fully independent. As a result, it’s unlikely that these
interfaces could be meaningfully separated or implemented in isolation.

This change consolidates them into a single `TaskManager` interface,
which can be reused where needed. The goal is to reduce the number of
concepts and components required to reason about the Ray core, and to
simplify the overall design.

Test:
- CI

Signed-off-by: Cuong Nguyen <can@anyscale.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

cszhu committed:
Updated stalebot to add bounced label and exempt labels (#54318)

We've had some stalebot conflicts where PRs are being marked stale
repeatedly. This is so that if a PR is marked "unstale", it will add a
`bounced` label. If a PR has this label, it will never be marked stale
again.

Signed-off-by: Christina Zhu <christina@anyscale.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

zcin committed:
[serve] deflake test_multiplex (#54335)

## Why are these changes needed?

Deflake `test_multiplex`.

https://buildkite.com/ray-project/postmerge/builds/11232#0197cd9b-97d6-4c15-9d3e-8740c601ecef/177-1298

Waiting for the `record_handle.add.remote` call to go through in
`__del__`.

---------

Signed-off-by: Cindy Zhang <cindyzyx9@gmail.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

ccmao1130 committed:
update daft package

Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

ccmao1130 committed:
update daft ver

Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

TimothySeah committed:
[train] Add broadcast_from_rank_zero and barrier collectives (#54066)

Add lightweight framework-agnostic `barrier` and
`broadcast_from_rank_zero` collectives for passing small data around
within ray train functions. Useful when we want all train workers to use
state that is only known at runtime (e.g. each worker saves a unique
file for the current epoch/step) and the framework doesn't support the
collective (e.g. lightgbm has limited support).

---------

Signed-off-by: Timothy Seah <tseah@anyscale.com>
Co-authored-by: Timothy Seah <tseah@anyscale.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

alexeykudinkin committed:
[Data] Fix flaky `test_shuffle` (#54339)

<!-- Thank you for your contribution! Please review
https://github.com/ray-project/ray/blob/master/CONTRIBUTING.rst before
opening a pull request. -->

<!-- Please add a reviewer to the assignee section when you create a PR.
If you don't have the access to it, we will shortly find a reviewer and
assign them to your PR. -->

## Why are these changes needed?

Removing byte size assertion that's not accurate anyway.

## Related issue number

<!-- For example: "Closes #1234" -->

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git
commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for
https://docs.ray.io/en/master/.
- [ ] I've added any new APIs to the API Reference. For example, if I
added a
method in Tune, I've added it in `doc/source/tune/api/` under the
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a
few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

---------

Signed-off-by: Alexey Kudinkin <ak@anyscale.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

zcin committed:
[serve] configure http options in controller (#54331)

We set a bunch of defaults for http and grpc options in the proxy, but
we should set them in the controller.

---------

Signed-off-by: Cindy Zhang <cindyzyx9@gmail.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

xinyuangui2 committed:
[Train] Remove the subclass relationship between RunConfig and RunConfigV1 (#54293)

In RunConfig, it subclasses RunConfigV1, whose `__post_init__` is using
v1 configs to initialize.
Now we remove the subclass relationship to avoid unexpected bugs.

---------

Signed-off-by: xgui <xgui@anyscale.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

elliot-barn committed:
raydepsets scaffolding (package management tool)  (#54265)

Scaffolding for raydepsets

Bazel build file contains:

- py_library: raydepsets_lib
- py_binary: raydepsets (cli binary)
- py_test: test_cli

click CLI (no functionality)

---------

Signed-off-by: elliot-barn <elliot.barnwell@anyscale.com>
Co-authored-by: Lonnie Liu <95255098+aslonnie@users.noreply.github.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

aslonnie committed:
[wheel] mac: upgrade arm64 wheel to macos 12 (#54323)

now wheels for both intel and apple silicon are on the same version

`ray[all]` does not work on arm64 macos 11 anyways, as `scipy` does not
have the related wheels

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

aslonnie committed:
[cpp] add explicit files for deps (#54311)

also make `ray_cpp_pkg` private visibility

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

dstrodtman reviewed (approved):
lgtm from the docs side.

dayshah committed:
[core] Don't try to monitor zipped files (#53151)

## Why are these changes needed?
Some users may implement their own log rotation on autoscaler event
files. If zipping the files during rotation, the event agent will still
try to read zipped files and raise an exception causing that file
monitor to fail. We try to open the files every 0.1 seconds. We only
need to look for new events in .log files that are actively being
written to.

This pr prevents opening and looking for changes in zipped files. There
will be no further changes in these files, so we don't need to monitor
them. This can lead to exceptions like
```
Exception: Read event file failed: /tmp/ray/session_2025-04-30_09-40-39_975641_2270/logs/events/event_AUTOSCALER.log.2.gz
    raise Exception(f"Read event file failed: {file}") from e
File "/home/ray/anaconda3/lib/python3.12/site-packages/ray/dashboard/modules/event/event_utils.py", line 176, in
_read_monitor_file
```

### Future state
Long term we'll kill this event agent / head entirely with the new
observability stack, and Ray will stop writing events to files. Any
events should follow a similar path to the new task events. An rpc with
events should be sent to the gcs from the component emitting the events
and the events will eventually get persisted in some storage, users can
plug-in their own storage to persist.

---------

Signed-off-by: dayshah <dhyey2019@gmail.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

rueian committed:
[core][refactor] move NodeManager::KillWorker to WorkerInterface::KillAsync for better testability (#54068)

Following the suggestion in
https://github.com/ray-project/ray/pull/53562#discussion_r2157785191,
this PR moves `NodeManager::KillWorker` to `WorkerInterface::Kill` so
that we can mock the method for testing if it is invoked or not, instead
of spawning a real process to see if it is killed or not.

As a side effect, this will also eliminate the confusion between
`NodeManager::KillWorker` and `NodeManager::DestroyWorker`.

---------

Signed-off-by: Rueian <rueian@anyscale.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

Qiaolin-Yu committed:
[core][GPU Objects] Add related tests for tensordict (#54286)

The current code already supports extracting tensors from `tensordict`,
and using external transport to transfer the tensors. This PR adds some
tests to prove this (e.g., checking the extracted tensor stored in the
gpu object manager, and testing nested tensordict).

Closes #52340.

Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

goutamvenkat-anyscale committed:
[Data] - write_parquet enable both partition by & min_rows_per_file, max_rows_per_file (#53930)

Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

aslonnie committed:
[deps] remove time series and ludwig dependencies (#54316)

not used anywhere

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

dayshah committed:
[core] Get cloud provider with ray on kubernetes (#51793)

On GKE
```
dhyey@cloudshell:~ (dhyey-dev)$ kubectl exec -it $HEAD_POD -- python -c "import requests; print(requests.get('http://metadata.google.internal/computeMetadata/v1',headers={'Metadata-Flavor': 'Google'}))"
Defaulted container "ray-head" out of: ray-head, autoscaler
<Response [200]>
dhyey@cloudshell:~ (dhyey-dev)$ kubectl exec -it $HEAD_POD -- python -c "import requests; print(requests.get('http://169.254.169.254/latest/meta-data/'))"
Defaulted container "ray-head" out of: ray-head, autoscaler
<Response [404]>
dhyey@cloudshell:~ (dhyey-dev)$ kubectl exec -it $HEAD_POD -- python -c "import requests; print(requests.get('http://169.254.169.254/metadata/instance?api-version=2021-02-01'))"
Defaulted container "ray-head" out of: ray-head, autoscaler
<Response [404]>
```

On anyscale on eks (google metadata req results in ConnectionError)
```
>>> print(requests.get('http://169.254.169.254/latest/meta-data/'))
<Response [200]>
>>> print(requests.get('http://169.254.169.254/metadata/instance?api-version=2021-02-01'))
<Response [404]>
```

Note: Untested on azure

---------

Signed-off-by: dayshah <dhyey2019@gmail.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

zcin committed:
[serve] deflake `test_autoscaling_policy` (#54336)

Increase the timeout in `test_e2e_raise_min_replicas` when waiting for
the first replica to start. Especially if there are multiple tests
running in parallel in CI, starting new replicas can be much slower.

example failure:

https://buildkite.com/ray-project/postmerge/builds/11251#0197d10c-7726-414b-8f62-8e507e887eb7/177-3331

Signed-off-by: Cindy Zhang <cindyzyx9@gmail.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

kouroshHakha committed:
[data.llm] Log engine stats after each batch task is done. (#54360)

Signed-off-by: Kourosh Hakhamaneshi <kourosh@anyscale.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

abrarsheikh committed:
reduce number of loops over request headers from 2 to 1 (#54326)

`__getitem__` on headers is a O(n) operation. Optimize the code to cut
out the extra loop.

---------

Signed-off-by: abrar <abrar@anyscale.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

aslonnie committed:
[ci] sync versions of pytest and pip-tools (#54315)

sync to the versions used in lock files

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

kouroshHakha committed:
[data.llm] Decouple max_tasks_in_flight from max_concurrent_batches (#54362)

Signed-off-by: Kourosh Hakhamaneshi <kourosh@anyscale.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

TimothySeah committed:
[train] Document ray.train.collective (#54340)

Signed-off-by: Timothy Seah <tseah@anyscale.com>
Co-authored-by: Timothy Seah <tseah@anyscale.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

edoakes committed:
[core] Deflake `test_actor_scheduling_not_block_with_placement_group` (#54368)

This test was flaking in CI due to a timeout:
https://buildkite.com/ray-project/postmerge/builds/11274#0197d7e7-a9ec-43fa-9b29-8561fbb34ec0/177-1584

As it was written, it was creating a 20 node cluster, which is far too
stressful to be in CI. I've reduced the stressfulness of the workload to
hopefully deflake, along with some other cleanup:

- Improved the test name and header comment.
- Removed the unnecessary `repeat(3)` that was burning CI time.
- Used `pg.wait()` to unify the checks for actor and PG readiness.
- Removed the captured nonlocal variable check.

---------

Signed-off-by: Edward Oakes <ed.nmi.oakes@gmail.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

edoakes committed:
[core] Deflake `test_reconstruction_suppression` (#54366)

This test flakes in CI due to transient `ActorUnavailableError`
(expected):
https://buildkite.com/ray-project/postmerge/builds/11276#0197d8fa-8bfe-4d64-9422-be76cffedc02/177-1290

Reworked the test to:

- improve the name
- retry the `ActorUnavailableError`
- make the behavior deterministic
- reduce the number of excess nodes.

---------

Signed-off-by: Edward Oakes <ed.nmi.oakes@gmail.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

TimothySeah committed:
[train] Force abort on SIGINT spam and do not abort finished runs (#54188)

Before this PR, spamming Ctrl C used to continue blocking until the
controller awaited. This was especially slow during worker group
initialization. Now, 3 Ctrl C's will force kill the run.

Before this PR, there was an edge case in which if you aborted a train
run after the controller set the state to finished/error but before it
exited, you would override a finished run to aborted. This is rare in
practice but it's good to cover all our bases.

---------

Signed-off-by: Timothy Seah <tseah@anyscale.com>
Co-authored-by: Timothy Seah <tseah@anyscale.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

aslonnie committed:
[data] dask: mark all dask-on-ray tests as manual (#54371)

have to run on python 3.12, and not sure what changes are required to
make them passing

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

edoakes committed:
[core] Remove `test_get_locations_timeout` (#54367)

This test sometimes fails because the location info _is_ available:
https://buildkite.com/ray-project/postmerge/builds/11255#0197d1e5-5ee0-4b16-b852-4ee17b39cc10

The entire implementation is problematic, and after data
[migrates](https://github.com/ray-project/ray/pull/53942) to
`get_local_object_locations`, `get_object_locations` will be deleted. So
for now deleting the flaky test.

---------

Signed-off-by: Edward Oakes <ed.nmi.oakes@gmail.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

rueian committed:
[core][autoscaler] make the autoscaler v2 work with the cluster launcher (#54230)

## Why are these changes needed?

This PR wires up the `NodeProviderAdapter` and `ThreadedRayInstaller` to
make autoscaler v2 work with the cluster launcher.

An autoscaler v2 enabled cluster can be launched by the following
command:

```sh
RAY_enable_autoscaler_v2=1 ray up aws.yaml -y
```

and this `aws.yaml`:

```yaml
cluster_name: default
max_workers: 2
upscaling_speed: 1.0
docker:
    image: "rayproject/ray:nightly-py312"
    container_name: "ray_container"
    pull_before_run: True
    run_options:
        - --ulimit nofile=65536:65536
idle_timeout_minutes: 5
provider:
    type: aws
    region: us-west-2
    availability_zone: us-west-2a,us-west-2b
    cache_stopped_nodes: True
auth:
    ssh_user: ubuntu
available_node_types:
    ray.head.default:
        resources: {}
        node_config:
            InstanceType: m5.large
            ImageId: ami-0387d929287ab193e
            BlockDeviceMappings:
                - DeviceName: /dev/sda1
                  Ebs:
                      VolumeSize: 140
                      VolumeType: gp3
    ray.worker.default:
        min_workers: 1
        max_workers: 2
        resources: {}
        node_config:
            InstanceType: m5.large
            ImageId: ami-0387d929287ab193e
            InstanceMarketOptions:
                MarketType: spot
head_node_type: ray.head.default

file_mounts: {
    "/home/ray/anaconda3/lib/python3.12/site-packages/ray/autoscaler/v2": "python/ray/autoscaler/v2",
}

cluster_synced_files: []
file_mounts_sync_continuously: False
rsync_exclude:
    - "**/.git"
    - "**/.git/**"
rsync_filter:
    - ".gitignore"
initialization_commands: []
setup_commands: []
head_setup_commands: []
worker_setup_commands: []
head_start_ray_commands:
    - ray stop
    - ray start --head --port=6379 --object-manager-port=8076 --autoscaling-config=~/ray_bootstrap_config.yaml --dashboard-host=0.0.0.0
worker_start_ray_commands:
    - ray stop
    - ray start --address=$RAY_HEAD_IP:6379 --object-manager-port=8076
```

The equivalent setup for gcp is also tested.

Note that the `image: "rayproject/ray:nightly-py312"` and `file_mounts`
are **required**, and the `python/ray/autoscaler/v2` local path should
point to this PR since we need to patch the autoscaler in the docker
image.

## Summary of Changes

* Make `ray up` accept the `RAY_enable_autoscaler_v2` env.
* Enable the `ThreadedRayInstaller` in the autoscaler v2 if the node
provider is a `NodeProviderAdapter`.
* Fix the `ThreadedRayInstaller` for not conflict with the autoscaler
v2.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git
commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for
https://docs.ray.io/en/master/.
- [ ] I've added any new APIs to the API Reference. For example, if I
added a
method in Tune, I've added it in `doc/source/tune/api/` under the
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a
few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

---------

Signed-off-by: Rueian <rueian@anyscale.com>
Signed-off-by: Rueian <rueiancsie@gmail.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

dayshah committed:
[core] Delete event_label (#54378)

event_label is just macros mapping to strings that are only used in one
spot. There really isn't a need for this, string literals are compile
time constants anyways.

Signed-off-by: dayshah <dhyey2019@gmail.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

edoakes committed:
[core] Add deprecation warning and remove tests for `_max_cpu_fraction_per_node` (#54380)

Per discussion in: https://github.com/ray-project/ray/pull/53610, this
argument was introduced for Ray AIR but is no longer used by any
libraries.

Out of an abundance of caution, adding a warning for a release before
removing it.

Deleting the integration tests eagerly since they burn CPU time in CI.

---------

Signed-off-by: Edward Oakes <ed.nmi.oakes@gmail.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

edoakes committed:
[core] Unskip `test_placement_group_strict_spread` (#54381)

Reason it was skipped seems to no longer be relevant:
https://github.com/ray-project/ray/issues/38726

Let's see if it flakes...

Signed-off-by: Edward Oakes <ed.nmi.oakes@gmail.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

alexeykudinkin committed:
[Data] Revisit async UDF handling in Ray Data (#54190)

<!-- Thank you for your contribution! Please review
https://github.com/ray-project/ray/blob/master/CONTRIBUTING.rst before
opening a pull request. -->

<!-- Please add a reviewer to the assignee section when you create a PR.
If you don't have the access to it, we will shortly find a reviewer and
assign them to your PR. -->

## Why are these changes needed?

Currently async UDFs handling diverges substantially from sync UDF
handling:

1. Sync UDFs: process 1 batch at a time in a loop, yielding results
using generator (generator will block if executor is not keeping up to
taking up blocks from it)
2. Async UDFs: launches 1 async task per batch and runs all of them
*concurrently*, adding produced results into the queue, while elements
of that queue are yielded back as a result.

Currently, In async UDF scenario concurrency is not constrained at all
-- meaning that:

1. Async task for every batch will be launched ASAP
2. Output queue is not constrained and will be growing unbounded
(potentially resulting into OOMs)
3. No back-pressure is applied to throttle production irrespective of
the speed of the consuming Ray Task

Changes
---

This PR addresses all of the aforementioned problems by revisiting async
UDF handling protocol:

1. Limiting # of batches handled concurrently to no more than specified
`max_concurrency` (inside `ray_remote_args` or defaults of 2 for
map_batches and 16 for row-based operations)
2. Fixing handling of coroutines vs async-generators
3. Adding comprehensive tests

## Related issue number

<!-- For example: "Closes #1234" -->

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git
commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for
https://docs.ray.io/en/master/.
- [ ] I've added any new APIs to the API Reference. For example, if I
added a
method in Tune, I've added it in `doc/source/tune/api/` under the
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a
few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

---------

Signed-off-by: Alexey Kudinkin <ak@anyscale.com>
Signed-off-by: Linkun <github@lkchen.net>
Co-authored-by: Linkun <github@lkchen.net>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

kouroshHakha committed:
[data.llm] Return a batch of rows in the udf instead of row by row (#54329)

Signed-off-by: Kourosh Hakhamaneshi <kourosh@anyscale.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

Qiaolin-Yu committed:
[core][GPU Objects] Disable tensordict tests in macos ci (#54375)

tensordict requires macos Sequoia to run, which version is too high to run on ray's macos CI

Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

khluu committed:
[release] Make KubeRay test run nightly (#54243)

Change KubeRay variant of this test to run nightly.
Please hold off on merging PR, @khluu needs to confirm if the job needs
to be deleted after each run (otherwise if the job stays on the cluster,
new jobs wouldn't run)

---------

Signed-off-by: Kevin H. Luu <kevin@anyscale.com>
Signed-off-by: kevin <kevin@anyscale.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

elliot-barn committed:
updating raydepsets test job (#54387)

Updating test job for raydepsets

- removing --only-tags and parallelism-per-worker flags

Signed-off-by: elliot-barn <elliot.barnwell@anyscale.com>
Co-authored-by: Lonnie Liu <95255098+aslonnie@users.noreply.github.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

aslonnie committed:
[ci] upgrade rayci version to 0.16.0 (#54392)

with some more features and goodies

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

alexeykudinkin committed:
[Data] Fixing `sort_benchmark` to avoid offsets overflows with Pyarrow (#54390)

<!-- Thank you for your contribution! Please review
https://github.com/ray-project/ray/blob/master/CONTRIBUTING.rst before
opening a pull request. -->

<!-- Please add a reviewer to the assignee section when you create a PR.
If you don't have the access to it, we will shortly find a reviewer and
assign them to your PR. -->

## Why are these changes needed?

Addresses https://github.com/ray-project/ray/issues/53546

Currently, we're using unbounded size data type (binary) that internally
utilizes int32 offsets to index individual elements. Using int32 offsets
imposes a max limit on the cumulative size of the column being 2^31-1
bytes, which this tests consistently violates.

This change addresses that by replacing arbitrary length `binary` type
with fixed-size `binary` type that doesn't use int32 offsets (instead
just relying on int64 element counter)

## Related issue number

<!-- For example: "Closes #1234" -->

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git
commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for
https://docs.ray.io/en/master/.
- [ ] I've added any new APIs to the API Reference. For example, if I
added a
method in Tune, I've added it in `doc/source/tune/api/` under the
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a
few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

---------

Signed-off-by: Alexey Kudinkin <ak@anyscale.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

aslonnie committed:
[data] dask: disable smoke test (#54411)

forgot to disable it last round

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

aslonnie committed:
[ci] release tests: remove unused app configs (#54410)

many release configs are not used anywhere anymore. the tests are no
longer using the buildfarm to build the image

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

harshit-anyscale committed:
migrate signature from _private to _common (#54357)

Fixes: https://github.com/ray-project/ray/issues/53478

- migrate ray._private.signature from `_private` to `_common`

Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

dayshah committed:
[core] Delete skip_flaky_core_test_premerge (#54382)

`skip_flaky_core_test_premerge` was always skipping all the tests, not
just on premerge

Signed-off-by: dayshah <dhyey2019@gmail.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

owenowenisme committed:
[Core] Use smart pointer in logging.cc (#54351)

We shouldn't use raw pointer so I changed to smart pointer in loggin.cc.
Also use raw string literals to replace escape quotes in the same file
for better readability.

Signed-off-by: You-Cheng Lin (Owen) <mses010108@gmail.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

dayshah committed:
[core] Inject reorder_wait_seconds for scheduling queue test (#54404)

This scheduling queue unit test would take >30 seconds before because
the default value of reorder_wait_seconds is 30. Now just injecting it
in one level higher so can inject it into the test as 1 second.

Signed-off-by: dayshah <dhyey2019@gmail.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

edoakes committed:
[serve] Remove usage of `ray._private.state` (#54140)

Replace with `ray.util.state`.

There is one remaining:
```
(ray) eoakes@eoakes-CQTYX0Y0RV serve % grep -r "_private.state" *
_private/cluster_node_info_cache.py:            ray._private.state.available_resources_per_node()
```

---------

Signed-off-by: Edward Oakes <ed.nmi.oakes@gmail.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

edoakes committed:
[serve] Remove usage of `internal_api.memory_summary()` (#54417)

Replace with usage of the `list_objects` public state API.

---------

Signed-off-by: Edward Oakes <ed.nmi.oakes@gmail.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

srinathk10 committed:
[Data] Add `pin_memory` to `iter_torch_batches` (#53792)

Add `pin_memory` to `iter_torch_batches`
- Whether to pin the memory of the tensors. Defaults to False.

---------

Signed-off-by: Srinath Krishnamachari <srinath.krishnamachari@anyscale.com>
Signed-off-by: srinathk10 <68668616+srinathk10@users.noreply.github.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

edoakes committed:
[core] Skip `test_owner_assign_inner_object` on Windows (#54383)

Test is flaky on Windows. `_owner=` arg is experimental and will be
deprecated/removed soon, no need to spend time debugging this on
Windows.

Closes https://github.com/ray-project/ray/issues/54333

Signed-off-by: Edward Oakes <ed.nmi.oakes@gmail.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

noemotiovon committed:
[Refactor]Rename NCCL-related items to comm_backend (#51061)

## Why are these changes needed?

### Background
This PR is a follow-up to
[#51032](https://github.com/ray-project/ray/pull/51032), which
introduced multi-device support in the Compiled Graph by leveraging
CUDA's NCCL backend for efficient out-of-band tensor communication.

While the current implementation is tightly coupled with NCCL and CUDA,
the Compiled Graph runtime is now ready to support a broader spectrum of
device types and collective communication backends (e.g., HCCL, RCCL).

### What This PR Does?
To enable extensibility and backend-agnostic design, this PR introduces
the following core changes:

Refactored NCCL-specific naming and APIs
NCCL-related modules, classes, and function names have been generalized
to eliminate hardcoded CUDA/NCCL assumptions.

Introduced a pluggable communication backend interface
A unified abstraction layer is added to decouple collective
communication logic from any specific implementation. This makes it
easier to support alternative collective libraries and device types in
the future.

This refactor does not alter the existing behavior of NCCL-based
Compiled Graph execution. All current workflows using CUDA+NCCL continue
to function as before.

## Related issue number
https://github.com/ray-project/ray/issues/51574

<!-- For example: "Closes #1234" -->

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git
commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for
https://docs.ray.io/en/master/.
- [ ] I've added any new APIs to the API Reference. For example, if I
added a
method in Tune, I've added it in `doc/source/tune/api/` under the
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a
few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

---------

Signed-off-by: noemotiovon <757486878@qq.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

ryanaoleary committed:
[Autoscaler][V2] Use running node instances to rate-limit upscaling (#50414)

This PR changes the v2 autoscaler scale up logic to use
`IMInstance.RAY_RUNNING` (rather than `IMInstance.REQUESTED` + nodes
with an `IMInstance.cloud_instance_id`) when computing the number of
nodes to launch per node type. This is to maintain consistency with how
`upscaling_speed` is handled by the V1 autoscaler and described in the
[Ray
docs](https://docs.ray.io/en/latest/cluster/vms/user-guides/configuring-autoscaling.html#upscaling-and-downscaling-speed),
i.e. the factor by which to scale the number of nodes in the cluster,
with a minimum of 5.

---------

Signed-off-by: Ryan O'Leary <ryanaoleary@google.com>
Co-authored-by: Kai-Hsun Chen <kaihsun@anyscale.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

aslonnie committed:
[wheel] remove pyarrow <18 restriction (#54405)

the restriction should be on installation time rather than inside the
wheel. this extra constraint sometimes can unnecessarily mislead
depedency manager's resolution algorithms

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

zcin committed:
[serve] bump timeout for test_proxy (#54426)

Move `test_proxy` to `medium` size test.

Signed-off-by: Cindy Zhang <cindyzyx9@gmail.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

aslonnie committed:
[deps] sync ml byod deps with global lock file (#54424)

update the versions of more-itertools and stuff

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

aslonnie committed:
[ci] add workspace status script (#54398)

and use it to reject '@' character in `HOME` and `USER`

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

zcin committed:
[serve] add request metadata to can_accept_request (#54429)

Add `request_metadata` as a parameter for future use.

Signed-off-by: Cindy Zhang <cindyzyx9@gmail.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

raulchen committed:
[data] Extract backpressure-related code from ResourceManager as a policy (#54376)

* Extract backpressure related methods (`can_submit_new_tasks` and
`max_task_output_bytes_to_read`) from ResourceManager, as a standalone
policy `ResourceBudgetBackpressurePolicy`)
* Report task_output_backpressure_time metric.

---------

Signed-off-by: Hao Chen <chenh1024@gmail.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

TimothySeah committed:
[train][doc] ray.train.report api docs should mention optional checkpoint_dir_name (#54391)

---------

Signed-off-by: Timothy Seah <tseah@anyscale.com>
Signed-off-by: Timothy Seah <timothy.seah777@yahoo.com>
Co-authored-by: Timothy Seah <tseah@anyscale.com>
Co-authored-by: Justin Yu <justinvyu@anyscale.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

vigneshka committed:
Update google tag container id (#54444)

<!-- Thank you for your contribution! Please review
https://github.com/ray-project/ray/blob/master/CONTRIBUTING.rst before
opening a pull request. -->

We don't have access to the previous tag manager and can't edit event
collection. This new one works

<!-- Please add a reviewer to the assignee section when you create a PR.
If you don't have the access to it, we will shortly find a reviewer and
assign them to your PR. -->

## Why are these changes needed?

<!-- Please give a short summary of the change and the problem this
solves. -->

## Related issue number

<!-- For example: "Closes #1234" -->

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git
commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for
https://docs.ray.io/en/master/.
- [ ] I've added any new APIs to the API Reference. For example, if I
added a
method in Tune, I've added it in `doc/source/tune/api/` under the
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a
few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

Signed-off-by: Vignesh Hirudayakanth <vigneshkanth@gmail.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

bveeramani committed:
[Data] Avoid OOMs with `read_json(..., lines=True)` (#54436)

<!-- Thank you for your contribution! Please review
https://github.com/ray-project/ray/blob/master/CONTRIBUTING.rst before
opening a pull request. -->

<!-- Please add a reviewer to the assignee section when you create a PR.
If you don't have the access to it, we will shortly find a reviewer and
assign them to your PR. -->

## Why are these changes needed?

<!-- Please give a short summary of the change and the problem this
solves. -->

`read_json(..., lines=True)` uses pandas to read batches of data. The
issue is that the batch size is always set to 10000 rows per batch, and
this can cause OOMs if each line is large (e.g., a line might contain
bytes for a 8 MiB image).

To avoid OOMs, this PR updates the implementation to sample a row and
infer the appropriate batch size.

## Related issue number

<!-- For example: "Closes #1234" -->

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git
commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for
https://docs.ray.io/en/master/.
- [ ] I've added any new APIs to the API Reference. For example, if I
added a
method in Tune, I've added it in `doc/source/tune/api/` under the
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a
few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

---------

Signed-off-by: Balaji Veeramani <bveeramani@berkeley.edu>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

srinathk10 committed:
Fix test_json CI failures (#54352)

<!-- Thank you for your contribution! Please review
https://github.com/ray-project/ray/blob/master/CONTRIBUTING.rst before
opening a pull request. -->

<!-- Please add a reviewer to the assignee section when you create a PR.
If you don't have the access to it, we will shortly find a reviewer and
assign them to your PR. -->

## Why are these changes needed?

<!-- Please give a short summary of the change and the problem this
solves. -->

Fix test_json CI failures

- Reduce the file size to KBs (instead of MBs) for testing file size
larger than block size in `test_json_read_file_larger_than_block_size`.

## Related issue number

<!-- For example: "Closes #1234" -->

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git
commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for
https://docs.ray.io/en/master/.
- [ ] I've added any new APIs to the API Reference. For example, if I
added a
method in Tune, I've added it in `doc/source/tune/api/` under the
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a
few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

---------

Signed-off-by: Srinath Krishnamachari <srinath.krishnamachari@anyscale.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

zcin committed:
[serve] skip test_proxy_disconnect_metrics on windows (#54441)

Skip `test_proxy_disconnect_metrics` on windows because it is flaky.

---------

Signed-off-by: Cindy Zhang <cindyzyx9@gmail.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

TimothySeah committed:
[train][checkpoint] Populate checkpoint from before_init_train_context instead of workergroupcontext (#54453)

This PR cleans up technical debt by implementing a TODO in the codebase.
The TODO was essentially to populate all Train Context args with
`before_init_train_context` methods.

Signed-off-by: Timothy Seah <tseah@anyscale.com>
Co-authored-by: Timothy Seah <tseah@anyscale.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

aslonnie committed:
[deps] upgrade datasets in release tests (#54425)

the older version of datasets does not with huggingface service any
more. `HfFilesystem` starts returning gzipped version of files, but the
client side (in `dataset`) is trying to decode it as an `utf-8` which
leads to failure.

also constrain myst-* deps with global lock, respects global lock file
versions on notebook testing

---------

Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

elliot-barn committed:
[ci] adding uv binary v0.7.19 (#54437)

Adding uv binary to be used in CI

---------

Signed-off-by: elliot-barn <elliot.barnwell@anyscale.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

lk-chen committed:
[llm] bump vllm to 0.9.2 (#54407)

Signed-off-by: Linkun <github@lkchen.net>
Signed-off-by: Linkun Chen <github@lkchen.net>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

Commit:
Add label selector observability to placement group tables and actor and task detail pages (#54292)

Follow-up to https://github.com/ray-project/ray/pull/53423

Missed a few places in the UI.
Also updates placement group tables to use the same code preview
component as the actor and tasks tables.

Placement group table
![Screenshot 2025-07-02 at 4 00
53 PM](https://github.com/user-attachments/assets/8de97470-abda-4680-b2fb-a4f90add0063)
![Screenshot 2025-07-02 at 4 00
56 PM](https://github.com/user-attachments/assets/a3c37e6f-c9db-4b37-b873-a5fbbd3012d7)

Actor detail
![Screenshot 2025-07-02 at 4 01
05 PM](https://github.com/user-attachments/assets/839cdaea-b441-4380-9c77-4ccb4ebfe563)

Task detail
![Screenshot 2025-07-02 at 4 01
19 PM](https://github.com/user-attachments/assets/aa12461c-a192-4114-a7fd-824613b9c6e6)

---------

Signed-off-by: Alan Guo <aguo@anyscale.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

edoakes committed:
[core] Delete old skipped tests (#54427)

See inline comments for each.

---------

Signed-off-by: Edward Oakes <ed.nmi.oakes@gmail.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

owenowenisme committed:
[Core] Use std::move in cluster task manager constructor (#54413)

Found that we pass by value in cluster task manager constructor, use
move to avoid unnecessary copy.

Signed-off-by: You-Cheng Lin (Owen) <mses010108@gmail.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

harshit-anyscale committed:
migrate check_library_usage_telemetry to _common (#54355)

Fixes: https://github.com/ray-project/ray/issues/53478

- migrating check_library_usage_telemetry from `_private` to `_common`
- migrating TelemetryCallsite from `_private` to `_common`.

Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

rueian committed:
[core][autoscaler] fix: use RAY_UP_enable_autoscaler_v2 instead of RAY_enable_autoscaler_v2 for ray up (#54456)

Use a different env var for ray up to enable autoscaler v2 to avoid
accidentally enabling v2 due to env inheritance.

Signed-off-by: Rueian <rueian@anyscale.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

DriverSong committed:
[serve.llm] Adaption of the change of vllm.PoolingOutput (#54467)

Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

Ziy1-Tan committed:
[Serve] Check multiple FastAPI ingress deployments in a single application (#53647)

## Why are these changes needed?

- Currently, Serve can not catch multiple FastAPI deployments in a
single application if user sets the docs path to None in their FastAPI
app.
- We can check multiple ASGIAppReplicaWrapper in a single application to
avoid this issue.

## Related issue number

Closes https://github.com/ray-project/ray/issues/53024

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git
commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for
https://docs.ray.io/en/master/.
- [ ] I've added any new APIs to the API Reference. For example, if I
added a
method in Tune, I've added it in `doc/source/tune/api/` under the
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a
few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

---------

Signed-off-by: Ziy1-Tan <ajb459684460@gmail.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

zcin committed:
[serve] deflake `test_autoscaling_policy_with_metr_disab` (#54458)

Deflake
`test_autoscaling_policy_with_metr_disab.py::TestAutoscalingMetrics::test_basic`

When `RAY_SERVE_COLLECT_AUTOSCALING_METRICS_ON_HANDLE=0`, we collect
ongoing request metrics at the replica and queued request metrics at the
handle -- but ongoing request metrics are updated very fast while queued
metrics are sent every 10s. Because of this delay the total number of
ongoing requests climbs to almost 100 because before the queued request
metrics are flushed, almost every request is double counted.

Example:
https://buildkite.com/ray-project/postmerge/builds/11322#0197eaca-62e1-457d-947b-a981210e98b9/177-852
Note that we are sending exactly 50 requests and expect the number of
replicas to scale to exactly 5. However the metrics grow above 50 here,
almost to 100, which causes the test to be flaky / fail.

This pr sets the env var `RAY_SERVE_HANDLE_METRIC_PUSH_INTERVAL_S=0.1`
and pairs with other stabilizing changes.

Signed-off-by: Cindy Zhang <cindyzyx9@gmail.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

landscapepainter committed:
[serve] update get_application_url and tests (#54449)

This PR replaces some of the manual string literals of urls within
`test_api`, `test_deploy`, `test_deploy_2`, `test_deploy_app`,
`test_failure` with `get_application_urls` and splits some of the tests
into separate files.

---------

Signed-off-by: doyoung <doyoung@anyscale.com>
Signed-off-by: Alexey Kudinkin <ak@anyscale.com>
Signed-off-by: Linkun <github@lkchen.net>
Signed-off-by: Kourosh Hakhamaneshi <kourosh@anyscale.com>
Signed-off-by: Kevin H. Luu <kevin@anyscale.com>
Signed-off-by: kevin <kevin@anyscale.com>
Signed-off-by: elliot-barn <elliot.barnwell@anyscale.com>
Signed-off-by: Lonnie Liu <lonnie@anyscale.com>
Signed-off-by: dayshah <dhyey2019@gmail.com>
Signed-off-by: You-Cheng Lin (Owen) <mses010108@gmail.com>
Signed-off-by: Edward Oakes <ed.nmi.oakes@gmail.com>
Signed-off-by: Srinath Krishnamachari <srinath.krishnamachari@anyscale.com>
Signed-off-by: srinathk10 <68668616+srinathk10@users.noreply.github.com>
Signed-off-by: noemotiovon <757486878@qq.com>
Signed-off-by: Ryan O'Leary <ryanaoleary@google.com>
Signed-off-by: Cindy Zhang <cindyzyx9@gmail.com>
Signed-off-by: Hao Chen <chenh1024@gmail.com>
Signed-off-by: Timothy Seah <tseah@anyscale.com>
Signed-off-by: Timothy Seah <timothy.seah777@yahoo.com>
Signed-off-by: Vignesh Hirudayakanth <vigneshkanth@gmail.com>
Signed-off-by: Balaji Veeramani <bveeramani@berkeley.edu>
Co-authored-by: Alexey Kudinkin <ak@anyscale.com>
Co-authored-by: Linkun <github@lkchen.net>
Co-authored-by: kourosh hakhamaneshi <31483498+kouroshHakha@users.noreply.github.com>
Co-authored-by: Qiaolin Yu <liin1211@outlook.com>
Co-authored-by: Kevin H. Luu <kevin@anyscale.com>
Co-authored-by: Elliot Barnwell <elliot.barnwell@anyscale.com>
Co-authored-by: Lonnie Liu <95255098+aslonnie@users.noreply.github.com>
Co-authored-by: harshit-anyscale <harshit@anyscale.com>
Co-authored-by: Dhyey Shah <dhyey2019@gmail.com>
Co-authored-by: Owen Lin (You-Cheng Lin) <106612301+owenowenisme@users.noreply.github.com>
Co-authored-by: Edward Oakes <ed.nmi.oakes@gmail.com>
Co-authored-by: srinathk10 <68668616+srinathk10@users.noreply.github.com>
Co-authored-by: Chenguang Li <757486878@qq.com>
Co-authored-by: Ryan O'Leary <113500783+ryanaoleary@users.noreply.github.com>
Co-authored-by: Kai-Hsun Chen <kaihsun@anyscale.com>
Co-authored-by: Cindy Zhang <cindyzyx9@gmail.com>
Co-authored-by: Hao Chen <chenh1024@gmail.com>
Co-authored-by: Timothy Seah <timothy.seah777@yahoo.com>
Co-authored-by: Timothy Seah <tseah@anyscale.com>
Co-authored-by: Justin Yu <justinvyu@anyscale.com>
Co-authored-by: Vignesh Hirudayakanth <vigneshkanth@gmail.com>
Co-authored-by: Balaji Veeramani <balaji@anyscale.com>
Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

omatthew98 reviewed (changes requested):
Looks like the microcheck failed because of the updated pip dependency? https://buildkite.com/ray-project/microcheck/builds/19843#0197f08b-3255-4fd3-b8b1-0c756dd4c467. Probably just need to recompile / update `requirements_compiled.txt`:  `requirements_compiled.txt is not up to date. Please download it from Artifacts tab and git push the changes.`

ccmao1130 committed:
requirements

Signed-off-by: ChanChan Mao <chanchanmao1130@gmail.com>

richardliaw commented:
hm looks like the commit history got messed up; open a new PR?
